File: ga/4omini_result_2024_7/result/1968C_cpp_mini/33_1968C.cpp
use std::io::{self, BufRead};

const N: usize = 500; 
const A: i32 = 1_000_000_000;

fn proc(n: usize, x: &[i32]) {
    let mut a = vec![0; N]; 
    a[0] = x[1] + 1;

    for i in 1..n {
        for j in 0..=A {
            a[i] = x[i] + j * a[i - 1];
            let next_x = if (i + 1) < n { x[i + 1] } else { 1 };
            if next_x < a[i] && a[i] != a[i - 1] {
                break;
            }
        }
    }

    for i in 0..n {
        print!("{} ", a[i]);
    }
    println!();
}

fn main() {
    let stdin = io::stdin();
    let reader = stdin.lock();
    let mut lines = reader.lines();
    
    let t: usize = lines.next().unwrap().unwrap().parse().unwrap();
    
    for _ in 0..t {
        let n: usize = lines.next().unwrap().unwrap().parse().unwrap();
        let mut x = vec![0; N];
        
        for i in 1..n {
            x[i] = lines.next().unwrap().unwrap().parse().unwrap();
        }

        proc(n, &x);
    }
}

File: ga/gpt3_5_result_2024_6/result/1968C_gpt/33_1968C.rs
use std::io::{self, BufRead};

const N: usize = 500;
const X: usize = 500;
const A: i32 = 1000000000;

fn proc() {
    let mut x: [i32; N] = [0; N];
    let mut a: [i32; N] = [0; N];

    let mut input = String::new();
    io::stdin().lock().read_line(&mut input).unwrap();
    let n: usize = input.trim().parse().unwrap();

    input.clear();
    io::stdin().lock().read_line(&mut input).unwrap();
    let mut parts = input.split_whitespace();
    for i in 1..n {
        x[i] = parts.next().unwrap().parse().unwrap();
    }

    a[0] = x[1] + 1;
    for i in 1..n {
        let mut j = 0;
        while j <= A {
            a[i] = x[i] + j * a[i - 1];
            if (i + 1 < n && x[i + 1] < a[i]) || a[i] != a[i - 1] {
                break;
            }
            j += 1;
        }
    }

    for i in 0..n {
        print!("{} ", a[i]);
    }
    println!();
}

fn main() {
    let stdin = io::stdin();
    let mut handle = stdin.lock();

    let mut input = String::new();
    handle.read_line(&mut input).unwrap();
    let t: usize = input.trim().parse().unwrap();

    for _ in 0..t {
        proc();
    }
}